pkg/agent/controller/egress/egress_controller.go

```diff
@@ -851,6 +874,7 @@ func (c *EgressController) syncEgress(egressName string) error {
 		if err := c.ipAssigner.AssignIP(desiredEgressIP, egress.Status.EgressNode != c.nodeName); err != nil {
 			return err
 		}
+		c.record.Eventf(egress, corev1.EventTypeNormal, "IPAssigned", "Assigned %s to %s", egress.Name, desiredNode)
```

Comment:
1. It shouldn't create a repeated event when the Egress is already on this Node, otherwise there would be too many events and most of them are just spam.
2. The event message should include the IP to be more informative.

Reply:
for the first part, when i tested only a single event is generated when the egress Ip is assigned to the node interface.
for the second part, sure I would add IP to the event message.

Reply:
I believe if you create a new Pod or delete a Pod (on the Node) that the Egress applies, it would generate an event.
And everytime antrea-agent restarts, it would generate an event.

Reply:
okay, will test these scenarios and based on the results will modify the code.

Reply:
Should we have e2e test for this feature? or manual test is enough? @tnqn

Reply:
Already made a suggestion regarding e2e test: https://github.com/antrea-io/antrea/pull/5765#pullrequestreview-1776967117